.github/workflows/pr-build.yaml

```diff
@@ -0,0 +1,237 @@
+name: PR build
+
+on:
+  pull_request:
+    branches: [main]
+
+env:
+  REGISTRY: ghcr.io
+  NAMESPACE: galasa-dev
+  IMAGE_TAG: main
+  
+jobs:
+  build-obr:
+    name: Build OBR using galasabld image and maven
+    runs-on: ubuntu-latest
+
+    steps:
+        - name: Checkout Code
+          uses: actions/checkout@v4
+          with:
+            path: obr
+      
+        - name: Checkout framework
+          uses: actions/checkout@v4
+          with:
+            repository: ${{env.NAMESPACE}}/framework
+            path: framework
+        
+        - name: Checkout managers
+          uses: actions/checkout@v4
+          with:
+            repository: ${{env.NAMESPACE}}/managers
+            path: managers
+    
+        - name: Checkout extensions
+          uses: actions/checkout@v4
+          with:
+            repository: ${{env.NAMESPACE}}/extensions
+            path: extensions
+        
+        - name: Set up JDK
+          uses: actions/setup-java@v4
+          with:
+            java-version: '11'
+            distribution: 'semeru'
+            cache: maven
+    
+        - name: Print githash
+          run: |
+            echo $GITHUB_SHA > ./obr.githash
+
+        - name: Make secrets directory
+          run : |
+              mkdir /home/runner/work/secrets
+
+        - name: Copy settings.xml
+          env:
+              MAVEN_SETTINGS_XML: ${{ secrets.MAVEN_SETTINGS_XML }}
+          run : |
+              echo $MAVEN_SETTINGS_XML > /home/runner/work/secrets/settings.xml
+        - name: Copy GPG passphrase
+          env:
+              GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
+          run : |
+              echo $GPG_PASSPHRASE > /home/runner/work/secrets/passphrase.file
+        - name: Copy GPG key
+          env:
+              GPG_KEY_BASE64: ${{ secrets.GPG_KEY }}
+          run : |
+              echo $GPG_KEY_BASE64 |  base64 --decode > /home/runner/work/secrets/galasa.gpg
+              
+        - name: Make GPG home directory and change permissions
+          run: |
+              mkdir /home/runner/work/gpg
+              chmod '700' /home/runner/work/gpg
+          
+        - name: Import GPG
+          run: |
+              gpg --homedir /home/runner/work/gpg --pinentry-mode loopback --passphrase-file /home/runner/work/secrets/passphrase.file --import /home/runner/work/secrets/galasa.gpg
+          
+        - name: Copy custom settings.xml
+          run: |
+              cp /home/runner/work/secrets/settings.xml /home/runner/work/gpg/settings.xml
+        
+        - name:  Pull and run the docker image

```

Comment:
These step names should reflect what is being built/generated here rather than the fact you are pulling the docker image to do it. Can we please change this to "Generate Galasa BOM"